### ISSUE TITLE
'sharedApplication' is unavailable - xcode compilation error with Firebase 4.11.0 & FirebaseAuth 4.5.0

### ISSUE THREAD

**Issue by jyaunches** [firebase/firebase-ios-sdk#1036]
* Xcode version: 9.2
  * Firebase SDK version: 4.11.0
  * Firebase Component: Auth 
  * Component version: 4.5.0

### [REQUIRED] Step 3: Describe the problem

I'm running a Swift 4.0, iOS 11 app attempting to upgrade to the new Firebase 4.11.0 & FirebaseAuth 4.5.0. 

My end purpose is to use Auth.auth().sendSignInLink capability, which is only available with these versions.

I'm getting the following compilation error inside the FirebaseAuth library:

![screen shot 2018-04-06 at 11 11 53 am](https://user-images.githubusercontent.com/372490/38429248-e4be2c7c-398b-11e8-922c-9b9198cc107c.png)

This is the relevant part of my Podfile

![screen shot 2018-04-06 at 11 14 44 am](https://user-images.githubusercontent.com/372490/38429281-f612b6fa-398b-11e8-80f9-103dfac714da.png)
:

Thank you for your speedy handling of github issues! You're doing a great job!

**Comment by protocol86** [comment]
@jyaunches,

Thanks for raising this issue. I have a quick question; are you building an App Extension?

cheers,
-Z

**Comment by jyaunches** [comment]
By 'App Extension', do you mean framework? If so, nope! This is just a standard iOS client app!

![screen shot 2018-04-07 at 1 29 21 pm](https://user-images.githubusercontent.com/372490/38458086-b6bec2be-3a67-11e8-9d84-0221d0d95292.png)

**Comment by paulb777** [comment]
From your screen shot, it looks like OneSignalNotificationServiceExtension is an extension. 

It's not valid to use UIApplication in an app extension and Firebase doesn't officially support app extensions.

**Comment by jyaunches** [comment]
You know.. your earlier question made me go back and look at the Podfile again. With the earlier version of the Cocoapod, I had moved it outside this target to make it available for the unit test suite. This may not be a good practice. Either way, I got things working by putting the pod specifications back inside the project target.

So, I'm good! Thanks for you time.

![screen shot 2018-04-07 at 3 28 34 pm](https://user-images.githubusercontent.com/372490/38459501-645f09be-3a78-11e8-86b6-f66421c7fd97.png)
